includes/class-wc-payments-order-service.php

```diff
@@ -1015,34 +1015,36 @@ private function mark_payment_authorized( $order, $intent_data ) {
 	/**
 	 * Updates an order to processing/completed status, while adding a note with a link to the transaction.
 	 *
-	 * @param WC_Order $order         Order object.
-	 * @param array    $intent_data   The intent data associated with this order.
+	 * @param WC_Order                          $order         Order object.
+	 * @param WC_Payments_API_Payment_Intention $intent        The intent instance.
 	 *
 	 * @return void
 	 */
-	private function mark_payment_capture_completed( $order, $intent_data ) {
-		$note = $this->generate_capture_success_note( $order, $intent_data['intent_id'], $intent_data['charge_id'] );
+	private function mark_payment_capture_completed( $order, $intent ) {
+		$intent_id     = $intent->get_id();
+		$fraud_outcome = $intent->get_metadata()['fraud_outcome'] ?? '';
+		$note          = $this->generate_capture_success_note( $order, $intent_id, $intent->get_charge()->get_id() );
 		if ( $this->order_note_exists( $order, $note ) ) {
 			return;
 		}
 
 		// Update the note with the fee breakdown details async.
-		$this->enqueue_add_fee_breakdown_to_order_notes( $order, $intent_data['intent_id'] );
+		$this->enqueue_add_fee_breakdown_to_order_notes( $order, $intent_id );
 
 		/**
 		 * If we have a status for the fraud outcome, we want to add the proper meta data.
 		 * If auth/capture is enabled and the transaction is allowed, it will be 'allow'.
 		 * If it was held for review for any reason, it will be 'review'.
 		 */
-		if ( '' !== $intent_data['fraud_outcome'] && Rule::is_valid_fraud_outcome_status( $intent_data['fraud_outcome'] ) ) {
+		if ( '' !== $fraud_outcome && Rule::is_valid_fraud_outcome_status( $fraud_outcome ) ) {
```

Comment:
Nit: we can place `$fraud_outcome = $intent->get_metadata()['fraud_outcome'] ?? '';` before this if-stement (better context bounding).

Reply:
Nice touch, done here https://github.com/Automattic/woocommerce-payments/pull/8107/commits/28d7730547aa740bdb808f17dac8fecf9457fda1 thanks